e2e/e2etests/test_rpc.go

```diff
@@ -0,0 +1,156 @@
+//
+// Usage modes:
+//  1. With tx hashes: Provide comma-separated transaction hashes as args to test specific transactions
+//     (useful for testing pre/post-upgrade transactions on live networks)
+//  2. Without args: Seeds test transactions and discovers existing transactions from this user via CCTX queries
+//     (both ZEVM and EVM chains) to test RPC methods against historical transactions
+//
+// This enables regression testing by validating RPC methods work correctly against transactions
+// submitted before upgrades, ensuring no breaking changes in the RPC layer.
+func TestRPC(r *runner.E2ERunner, args []string) {
+	r.Logger.Info("starting JSON-RPC tests")
+
+	rpcWrapper := rpcwrapper.NewZEVMRPC(r.ZEVMClient)
+
+	txHashes := []string{}
+	if len(args) == 1 && args[0] != "" {
+		// Parse comma-separated transaction hashes for testing specific transactions
+		// Example: "0x123...,0x456..." for testing pre/post-upgrade transactions
+		txHashes = strings.Split(args[0], ",")
+	}
+
+	if len(txHashes) == 0 {
+		txHashes = seedAndDiscoverUserTestTxs(r)
+	}
+
+	// testing just most important rpc methods, if there are some errors in rpc layer probably most will fail
+	// just testing sanity checks, like presence of fields, so this can be executed for provided hashes on livenet
+	for _, txHash := range txHashes {
+		receipt, err := rpcWrapper.EthGetTransactionReceipt(r.Ctx, common.HexToHash(txHash))
+		require.NoError(r, err)
+		require.NotNil(r, receipt)
+		require.NotEmpty(r, receipt.BlockHash)
+		require.NotEmpty(r, receipt.BlockNumber)
+		require.NotEmpty(r, receipt.TransactionHash)
+
+		txInfo, err := rpcWrapper.EthGetTransactionByHash(r.Ctx, common.HexToHash(txHash))
+		require.NoError(r, err)
+		require.NotNil(r, txInfo)
+		require.NotEmpty(r, txInfo.Hash)
+		require.Equal(r, txHash, txInfo.Hash)
+		require.NotEmpty(r, txInfo.From)
+		require.NotEmpty(r, txInfo.Nonce)
+		require.NotEmpty(r, txInfo.Gas)
+
+		blockNumber, err := hexutil.DecodeBig(receipt.BlockNumber)
+		require.NoError(r, err)
+
+		blockByNumber, err := rpcWrapper.EthGetBlockByNumber(r.Ctx, blockNumber, false)
+		require.NoError(r, err)
+		require.NotNil(r, blockByNumber)
+		require.Equal(r, receipt.BlockNumber, blockByNumber.Number)
+		require.NotEmpty(r, blockByNumber.Hash)
+		require.NotEmpty(r, blockByNumber.ParentHash)
+		require.NotEmpty(r, blockByNumber.Timestamp)
+		require.GreaterOrEqual(r, len(blockByNumber.Transactions), 1)
+
+		blockByHash, err := rpcWrapper.EthGetBlockByHash(r.Ctx, common.HexToHash(receipt.BlockHash), false)
+		require.NoError(r, err)
+		require.NotNil(r, blockByHash)
+		require.Equal(r, receipt.BlockHash, blockByHash.Hash)
+		require.NotEmpty(r, blockByHash.Number)
+		require.NotEmpty(r, blockByHash.ParentHash)
+		require.NotEmpty(r, blockByHash.Timestamp)
+		require.GreaterOrEqual(r, len(blockByHash.Transactions), 1)
+
+		traceTx, err := rpcWrapper.DebugTraceTransaction(r.Ctx, common.HexToHash(txHash))
```

Comment:
Does `DebugTraceTransaction` API relies on the state history? If yes, it has to be ran either on a archive node endpoint or within a given limit of state history length (by blocks).

Reply:
if not intended for live network at this moment, it's should be fine.